north/src/manifest.rs

```diff
@@ -160,15 +136,39 @@ pub enum MountFlag {
 }
 
 #[derive(Clone, Eq, PartialEq, Debug, Serialize, Deserialize)]
-pub struct Mount {
-    /// Mount source
-    pub source: Option<std::path::PathBuf>,
-    /// Mount target
-    pub target: std::path::PathBuf,
-    /// Mount type
-    pub r#type: MountType,
-    /// Mount flags,
-    pub flags: Option<HashSet<MountFlag>>,
+#[serde(untagged)]
+pub enum MountSource {
```

Comment:
This is just used for the deserialisation and could be declared private and within `deserialize_mounts`.